Source/WebCore/platform/graphics/FontCascade.h

```diff
@@ -233,6 +233,19 @@ class FontCascade final : public CanMakeWeakPtr<FontCascade>, public CanMakeChec
     // Keep this in sync with RenderText's m_fontCodePath
     enum class CodePath : uint8_t { Auto, Simple, Complex, SimpleWithGlyphOverflow };
     WEBCORE_EXPORT CodePath codePath(const TextRun&, std::optional<unsigned> from = std::nullopt, std::optional<unsigned> to = std::nullopt) const;
+
+    CodePath codePathForShaping(const TextRun& run, std::optional<unsigned> from = std::nullopt, std::optional<unsigned> to = std::nullopt) const
+    {
+#if ENABLE(COMPLEX_TEXT_CONTROLLER_FOR_SIMPLE_CODE_PATH)
+        UNUSED_PARAM(run);
+        UNUSED_PARAM(from);
+        UNUSED_PARAM(to);
+        return CodePath::Complex;
+#else
```

Comment:
Any reason why we don't simply add this fragment to codePath itself?

Reply:
That's the current approach. GTK and WPE ports are always choosing the complex text code path by doing `FontCascade::setCodePath(CodePath::Complex)`.

Reply:
Thanks for clarifying it!